File: java/problem_3.java
public class problem_3
{
	public static void main(String[] args)
	{
		long number = 600851475143L;

		for(long i = 2L; i<number; i++)
		{
			if(number%i==0 && i>2)
			{
				if(prime_check(i))
				{
					System.out.println("prime divisor: " +i);
				}
			}
		}


	}

	public static boolean prime_check(long number)
	{
		for(long i=2L; i<number; i++)
		{
			if(number%i==0)
			{
				return false;
			}
		}

		return true;
	}
}

File: javascript/problem_3.js
//problem 3

function prime_check(num)
{

	for (var n= 2; n <num; n++)
	{
		if(num %n=== 0)
		{	
			console.log("false");
			return false;
		}
	}

	console.log("true");
	return true;
}

function main(number)
{
	for(var x=2; x <number+1; x++ )
	{
		if(number%x===0 && prime_check(x))
		{
			console.log(x);
			var answer = x;
		}
	}

	console.log(answer);
	return answer;
}

main(600851475143);
